File: Java/src/com/dbc/algorithm/ReferOffer/BaShuZiFanYiChengZiFuChuanLcof.java
package com.dbc.algorithm.ReferOffer;

public class BaShuZiFanYiChengZiFuChuanLcof {
    private int res = 0;
    public int translateNum(int num) {
        String strNums = "" + num;
        dfs(strNums, 0, strNums.length());
        return res;
    }

    public void dfs(String strNum, int point, int length) {
        if (point == length) {
            res++;
            return;
        }
        if (point < length - 1 && strNum.charAt(point) != '0' && Integer.parseInt(strNum.substring(point, point + 2)) < 26) {
            dfs(strNum, point + 2, length);
        }
        dfs(strNum, point + 1, length);
    }
}

File: core/tmp/Python/refer-offer/ba_shu_zi_fan_yi_cheng_zi_fu_chuan_lcof.py
def translateNum(num: int) -> int:
    res = [0]
    strNum = str(num)
    length = len(strNum)
    def dfs(point: int):
        if point == length:
            res[0] += 1
            return
        if  point < length - 1 and strNum[point] != '0' and int(strNum[point:point + 2]) <= 25:
            dfs(point + 2)
        dfs(point + 1)
        
    dfs(0)
    return res[0]

if __name__ == "__main__":
    print(translateNum(506))
